perceval/algorithm/tomography/tomography.py

```diff
@@ -106,82 +45,68 @@ class StateTomography(AAlgorithm):
     - Adds preparation and measurement circuits to input processor (with the gate operation under study)
     - Computes parameters required to do state tomography
     - Performs Tomography experiment - Computes and Returns density matrices for each input state
-    """
-    def __init__(self, nqubit: int, operator_processor: Processor, post_process=False, renormalization=None):
-        super().__init__(processor=operator_processor)
-        self._nqubit = nqubit
-        self._operator_processor = operator_processor  # Gate operation under study
-        self._backend = operator_processor.backend  # default - SLOSBackend()
-        self._source = operator_processor.source  # default - ideal source
-        self._post_process = post_process
-        self._renormalization = renormalization
-        self._heralded_modes = [(key, value) for key, value in operator_processor.heralds.items()]
-        self._size_hilbert = 2 ** nqubit
-        # Todo: I need to put this number somewhere, i forgot. find [number of elements = between (0 -> 4**nqubits-1)]
-
-    def _input_state_dist_config(self):
-        # Configures the input state for the Processor
-        input_state = BasicState("|1,0>")
-        for _ in range(1, self._nqubit):
-            # setting the input state for the gate qubit modes
-            input_state *= BasicState("|1,0>")
-        for m in self._heralded_modes:
-            # setting the input for heralded modes of the given processor
-            input_state *= BasicState([m[1]])
-        input_distribution = self._source.generate_distribution(expected_input=input_state)
-        return input_distribution
 
-    def _compute_probs(self, prep_state_indices, meas_pauli_basis_indices):
+    :param operator_processor: Gate under study
+    """
+    def __init__(self, operator_processor: AProcessor, **kwargs):
+        super().__init__(processor=operator_processor, **kwargs)
+        self._nqubit, odd_modes = divmod(operator_processor.m, 2)
+        if odd_modes:
```

Comment:
Could you add a unit test showing that a 5 mode circuit does not trigger any computation?